database/src/main/java/org/akvo/flow/database/migration/GroupsMigrationHelper.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2021 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.database.migration
+
+import android.content.ContentValues
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import org.akvo.flow.database.SurveyColumns
+import org.akvo.flow.database.tables.QuestionGroupTable
+import org.akvo.flow.database.tables.Tables
+import org.akvo.flow.utils.FileHelper
+import org.akvo.flow.utils.XmlFormParser
+import timber.log.Timber
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.util.ArrayList
+
+class GroupsMigrationHelper {
+
+    fun migrateGroups(db: SQLiteDatabase, context: Context) {
+        val cursor =
+            db.query(Tables.SURVEY, arrayOf(SurveyColumns._ID), null, null, null, null, null)
+        val formIds = ArrayList<String>()
```

Comment:
Maybe this should be something like a `Set` to remove duplicates or is that already handled in the query?

Reply:
that's a good point